Gordon360/Services/LostAndFoundService.cs

```diff
@@ -0,0 +1,338 @@
+        public IEnumerable<MissingItemReportViewModel> GetMissingItems(string username)
+        {
+            // Get the account of the given username.
+            var account = context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username);
+
+            string idNum;
+
+            if (account != null)
+            {
+                idNum = account.gordon_id;
+            }
+            else
+            {
+                throw new ResourceCreationException() { ExceptionMessage = "No account could be found for the usename." };
+            }
+            // Fetch the missing item reports belonging to that user, and not submitted for someone else
+            IEnumerable<MissingItemData> missingList = context.MissingItemData.Where(x => x.submitterID == idNum && x.forGuest == false);
+            // Convert the report objects into viewmodel objects.
+            IEnumerable<MissingItemReportViewModel> returnList = missingList.Select(x => (MissingItemReportViewModel)x);
+
+            // Get the list of actions taken that are public, and add them to each missing item report.
+            List<IEnumerable<ActionsTakenViewModel>> actionsTakenList = [];
+            foreach (MissingItemReportViewModel item in returnList)
+            {
+                if (item.recordID != null)
+                {
+                    actionsTakenList.Add(GetActionsTaken((int)item.recordID, true));
+                }
+            }
+
+            IEnumerator<IEnumerable<ActionsTakenViewModel>> actionsTakenEnumerator = actionsTakenList.GetEnumerator();
+            // Mix-in actions taken lists for each report
+            returnList = returnList.Select(x => { actionsTakenEnumerator.MoveNext(); x.adminActions = actionsTakenEnumerator.Current; return x; });
+
+            return returnList;
+        }
+
+        /// <summary>
+        /// Get all missing item reports
+        /// </summary>
+        /// <returns>An enumerable of Missing Item Reports, from the Missing Item Data view</returns>
+        public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll()
+        {
+            IEnumerable<MissingItemData> missingList = context.MissingItemData.AsEnumerable();
+            IEnumerable<MissingItemReportViewModel>missingModelList = missingList.Select(x => (MissingItemReportViewModel)x);
+            List<IEnumerable<ActionsTakenViewModel>> actionsTakenList = [];
+            foreach (MissingItemReportViewModel item in missingModelList)
+            {
+                if (item.recordID != null)
+                {
+                    actionsTakenList.Add(GetActionsTaken((int)item.recordID));
+                }
+            }
+
+            IEnumerator<IEnumerable<ActionsTakenViewModel>> actionsTakenEnumerator = actionsTakenList.GetEnumerator();
+
+            missingModelList = missingModelList.Select(x => { actionsTakenEnumerator.MoveNext(); x.adminActions = actionsTakenEnumerator.Current; return x; });
+            return missingModelList;
```

Comment:
```suggestion
            IEnumerable<MissingItemData> missingItems = context.MissingItemData.Select(x => (MissingItemReportViewModel)x);

            foreach (MissingItemReportViewModel item in missingItems)
            {
                if (item.recordID != null)
                {
                    item.adminActions = GetActionsTaken((int)item.recordID);
                }
            }
            
            return missingItems;
```

Reply:
Tried writing the code to set item.adminActions=GetActionsTaken.  Unfortunately that doesn't actually seem to change the values in the missingItems object, I'm guessing item is not a reference to the actual object in the array, so we can't modify it like this.  Unless there's another better way, I'm going to keep the code using a missing item list, and select statement to combine them.  It's a bit convoluted, but that's the only method I could get to work.

Reply:
No, `item` is definitely a reference to the actual object in the array. Foreach is the standard way to mutate items in a collection in C#.

I'm realizing my snippet above had an issue, so if you copy-pasted that, you may have seen an odd behavior. Specifically, I left the type of `missingItems` as `IEnumberable<MissingItemData>`, rather than `IEnumerable<MissingItemReportViewModel>`. Since you are using the casting syntax, the data being stored in the collection is of type `MissingItemData`. However, in the foreach loop, the `item` variable is explicitly declared to be of type `MissingItemReportViewModel`, and since the implicit operator is defined for that conversion, I think the compiler is allocating a new MissingItemReportViewModel object for each item and casting the actual array elements data into that item.

You can easily show that [foreach mutates collection elements correctly](https://dotnetfiddle.net/I1QIIr). Can you paste the code that you're using here?

Reply:
Sure, here's the code I tried:
```
IEnumerable<MissingItemReportViewModel> missingItem = context.MissingItemData.Select(x => (MissingItemReportViewModel)x);

foreach (MissingItemReportViewModel item in missingItem)
{
    if (item.recordID != null)
    {
        item.adminActions = (GetActionsTaken((int)item.recordID, username));
    }
}

return missingItem;
```

But the resulting object in swagger looks like:

```
{
    "recordID": 127,
    "firstName": "Matthew",
    "lastName": "Jones",
    "category": "clothing/shoes",
    "colors": [
      "Black"
    ],
    "brand": "",
    "description": "New Test Item",
    "locationLost": "New Test Location",
    "stolen": false,
    "stolenDescription": "",
    "dateLost": "2024-12-05T00:00:00",
    "dateCreated": "2024-12-05T00:00:00",
    "phone": "9789790323",
    "email": "Matthew.Jones@gordon.edu",
    "status": "active",
    "submitterUsername": "Matthew.Jones",
    "submitterID": null,
    "forGuest": false,
    "adminActions": null
  }
```

With adminActions always being ```null```...  Not sure why?

Reply:
@EjPlatzer ah, that explains the poor performance already (which is bad since its only test data!).  But this seems like a much better solution for dealing with the related entities 👍.

Reply:
Resolved in API #1111, runs much faster now 👍!